### PR TITLE
TF-1613 Fix  when network is down, Tmail shouldn't prompt for a password

### PR DESCRIPTION
**Pull request by dab246**
### Issue

#1613 

### Resolved

https://user-images.githubusercontent.com/80730648/229799786-e18b48b1-7bfc-4d79-9719-6b6efaad5c85.mov

### CODE REVIEW

File: core/lib/presentation/utils/html_transformer/dom/image_transformers.dart [hunk 1]
```diff
@@ -23,20 +23,26 @@ class ImageTransformer extends DomTransformer {
...
+      if (src != null) {
+        log('ImageTransformer::process(): src: $src');
+        final cid = src.replaceFirst('cid:', '').trim();
+        final urlDownloadCid = mapUrlDownloadCID?[cid];
+        log('ImageTransformer::process(): urlDownloadCid: $urlDownloadCid');
+        if (urlDownloadCid?.isNotEmpty == true && dioClient != null) {
+          final imgBase64Uri = await loadAsyncNetworkImageToBase64(
+              dioClient,
+              compressFileUtils,
+              urlDownloadCid!);
```
Comment (nqhhdev) [inline core/lib/presentation/utils/html_transformer/dom/image_transformers.dart:42]:
```suggestion
          final imgBase64Uri = await loadAsyncNetworkImageToBase64(
            dioClient,
            compressFileUtils,
            urlDownloadCid!);
```

File: lib/features/composer/presentation/composer_controller.dart [hunk 2]
```diff
@@ -266,51 +266,44 @@ class ComposerController extends BaseController {
...
+    if (success is GetEmailContentLoading) {
+      emailContentsViewState.value = Right(success);
+    } else if (success is LocalFilePickerSuccess) {
+      _pickFileSuccess(success);
+    } else if (success is GetEmailContentSuccess) {
+      _getEmailContentSuccess(success);
+    } else if (success is GetAllIdentitiesSuccess) {
+      _handleGetAllIdentitiesSuccess(success);
+    } else if (success is DownloadImageAsBase64Success) {
+      if(kIsWeb) {
```
Comment (nqhhdev) [inline lib/features/composer/presentation/composer_controller.dart:280]:
```suggestion
      if (kIsWeb) {
```

File: lib/features/login/presentation/base_login_view.dart [hunk 3]
```diff
@@ -74,27 +74,23 @@ abstract class BaseLoginView extends GetWidget<LoginController> {
...
+                    side: const BorderSide(width: 0, color: AppColor.primaryColor)
+                ))
+            ),
+            child: Text(AppLocalizations.of(context).signIn,
+                style: const TextStyle(fontSize: 16, color: Colors.white)
+            ),
+            onPressed: () {
+              loginController.handleLoginPressed();
+            }
         )
```
Comment (nqhhdev) [inline lib/features/login/presentation/base_login_view.dart:94]:
```suggestion
        child: ElevatedButton(
          key: const Key('loginSubmitForm'),
          style: ButtonStyle(
            foregroundColor: MaterialStateProperty.resolveWith<Color>((Set<MaterialState> states) => Colors.white),
            backgroundColor: MaterialStateProperty.resolveWith<Color>(
              (Set<MaterialState> states) => AppColor.primaryColor),
            shape: MaterialStateProperty.all(RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(10),
              side: const BorderSide(width: 0, color: AppColor.primaryColor)))
          ),
          child: Text(AppLocalizations.of(context).signIn,
            style: const TextStyle(fontSize: 16, color: Colors.white)
          ),
          onPressed: () {
            loginController.handleLoginPressed();
          }
        ),
```

**Review by nqhhdev — APPROVED** [review]

**Comment by hoangdat** [comment]
- please test Logout in OIDC

**Comment by hoangdat** [comment]
IMO, you miss handle the error for this case. `get session interactor` not go through `consumeState()` so `onError` never be called.

image

**Comment by dab246** [comment]
> IMO, you miss handle the error for this case. `get session interactor` not go through `consumeState()` so `onError` never be called. It means that, when you throw any exception in repository layer, this interactor catches this exception and return Future value with Either.Left
> 
> image



> consumeState

We already use `consumeState` for `get session interactor`

**Comment by dab246** [comment]
Currently all the `Interactors` of the app are `return Stream`. Except for some interactors for getting the `autocomplete` data of the widget

**Comment by dab246** [comment]
> * please test Logout in OIDC


https://user-images.githubusercontent.com/80730648/230314122-0e108ce5-12aa-407e-9675-fbd09f6b7c62.mov

**Review by hoangdat — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Pull request merged**